homeassistant/components/lutron_caseta/config_flow.py

```diff
@@ -0,0 +1,29 @@
+"""Config flow for Lutron Caseta."""
+
+from homeassistant import config_entries
+
+from . import DOMAIN  # pylint: disable=unused-import
+
+ABORT_REASON_ALREADY_CONFIGURED = "already_configured"
+
+ENTRY_DEFAULT_TITLE = "Caseta Bridge"
+
+
+class LutronCasetaFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Hue config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_import(self, import_info):
+        """Import a new Caseta bridge as a config entry.
+
+        This flow is triggered by `async_setup`.
+        """
+
+        # Abort if existing entry with matching host exists.
+        host = import_info["host"]
+        if any(host == entry.data["host"] for entry in self._async_current_entries()):
+            return self.async_abort(reason=ABORT_REASON_ALREADY_CONFIGURED)
+
+        return self.async_create_entry(title=ENTRY_DEFAULT_TITLE, data={"host": host})
```

Comment:
Before creating the entry we need to check that we can create the bridge with the config and connect to the host.

We'll throw away this bridge instance and create a new one in `async_setup_entry`. Here it's just for checking and validating the config.

Reply:
@MartinHjelmare creating the bridge instance requires `keyfile`, `certfile`, etc., which currently are stored in `hass.data` keyed by `host` (`host` is currently the only thing stored in the config entry data).

Would you read from `hass.data` here, or just move the `keyfile`/`certfile`/etc. (everything required to create the bridge instance) to the config entry data?

It seems like the latter might be the wiser option, to make the config entries self-sufficient. Just wanted to verify whether I’ve got the right thinking.

Reply:
The config entry data must be json serializable. As long as that works, it sounds good to put this data in the config entry data.